src/test/java/org/apache/sysds/test/functions/builtin/BuiltinALSCGTest.java

```diff
@@ -0,0 +1,50 @@
+package org.apache.sysds.test.functions.builtin;
+
+import org.apache.sysds.test.AutomatedTestBase;
+import org.apache.sysds.test.TestConfiguration;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BuiltinALSCGTest extends AutomatedTestBase {
+
+    private final static String TEST_NAME = "als_cg";
+    private final static String TEST_DIR = "functions/builtin/";
+    private static final String TEST_CLASS_DIR = TEST_DIR + BuiltinALSCGTest.class.getSimpleName() + "/";
+
+    private final static int rows = 300;
+    private final static int cols = 20;
+
+    @Override
+    public void setUp() {
+        addTestConfiguration(TEST_NAME,new TestConfiguration(TEST_CLASS_DIR, TEST_NAME,new String[]{"B"}));
+    }
+
+    @Test
+    public void testALSCG() {
+        runtestALSCG();
+    }
+
+    private void runtestALSCG(){
+
+        loadTestConfiguration(getTestConfiguration(TEST_NAME));
+        String HOME = SCRIPT_DIR + TEST_DIR;
+        fullDMLScriptName = HOME + TEST_NAME + ".dml";
+        List<String> proArgs = new ArrayList<>();
+
+        proArgs.add("-explain");
+        proArgs.add("-stats");
```

Comment:
Once you are happy with the tests, then remove the -explain and -stats flag if you don't use their outputs.

Reply:
Will do that at the end.